x-pack/plugins/security_solution/public/detections/components/rules/data_view_selector/index.tsx

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState, useEffect } from 'react';
+
+import {
+  EuiCallOut,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFormRow,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { DataViewListItem } from '@kbn/data-views-plugin/common';
+import { DataViewBase } from '@kbn/es-query';
+import { FieldHook, getFieldValidityAndErrorMessage } from '../../../../shared_imports';
+import * as i18n from './translations';
+import { useKibana } from '../../../../common/lib/kibana';
+import { DefineStepRule } from '../../../pages/detection_engine/rules/types';
+
+interface DataViewSelectorProps {
+  kibanaDataViews: { [x: string]: DataViewListItem };
+  field: FieldHook<DefineStepRule['dataViewId']>;
+  setIndexPattern: (indexPattern: DataViewBase) => void;
+}
+
+export const DataViewSelector = ({
+  kibanaDataViews,
+  field,
+  setIndexPattern,
+}: DataViewSelectorProps) => {
```

Comment:
This is the component we're looking to replace with the new `DataViewPicker` from the `unified-search` plugin, correct?